frontend/src/components/App/Notifications/index.tsx

```diff
@@ -0,0 +1,226 @@
+import bellIcon from '@iconify/icons-mdi/bell';
+import { Icon } from '@iconify/react';
+import { Badge, Grid, IconButton, makeStyles, Theme, Tooltip, Typography } from '@material-ui/core';
+import { Menu, MenuItem } from '@material-ui/core';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { useHistory } from 'react-router';
+import { FixedSizeList } from 'react-window';
+import Event, { KubeEvent } from '../../../lib/k8s/event';
+import { Notification } from '../../../lib/notification';
+import { createRouteURL } from '../../../lib/router';
+import { setUINotifications, updateUINotification } from '../../../redux/actions/actions';
+import { useTypedSelector } from '../../../redux/reducers/reducers';
+import { DateLabel } from '../../common';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  notificationItem: {
+    '&.MuiMenuItem-root': {
+      borderBottom: `1px solid ${theme.palette.notificationBorderColor}`,
+      padding: '1rem',
+    },
+  },
+  notificationMessage: {
+    '&': {
+      wordBreak: 'break-word',
+      whiteSpace: 'normal',
+    },
+  },
+  root: {
+    '& .MuiPaper-root': {
+      width: '30vw',
+      minWidth: '300px',
+      maxHeight: '70vh',
+    },
+  },
+  errorItem: {
+    '&': {
+      color: theme.palette.error,
+    },
+  },
+}));
+
+function prepareNotificationsList(
```

Comment:
Just out of curiosity, why did you choose to make this a function and not a component?

Reply:
updated this, I actually at first made it a pure function but i think it is not a pure function because it has hook calls inside it.